pipelines/components/aggregate_android.py

```diff
@@ -7,20 +7,39 @@ def bigquery_aggregate_events_op(
 ) -> str:
     from google.cloud import bigquery
 
-    def aggregate_events_query():
+    def aggregate_events_query(select_statement: str):
         return f"""
                 INSERT INTO `{aggregation_table}`
-                SELECT currentMeasurement.*, current_hash as fields_hash, NULL, NULL FROM `{updates_table}`;
+                SELECT {select_statement} FROM `{updates_table}`;
                 DELETE FROM `{aggregation_table}`
                 WHERE (CONCAT(fields_hash, stats_timestamp) 
                 IN (SELECT CONCAT(previous_hash, previousMeasurement.stats_timestamp) concatenated_updates 
                 FROM `{updates_table}`))
         """
 
+    # dynamically build the select statement to appropriately follow any updates in the schema
+    def get_aggregate_select_statement():
+        project = aggregation_table.split(".")[1]
+        table = aggregation_table.split(".")[2]
+        table_ref = bigquery.DatasetReference(project_name, project).table(table)
+        struct_schema = bigquery_client.get_table(table_ref).schema
+        select_statement = []
+        for field in struct_schema:
+            if field.name in ["screen_display_scale_default_comparison", "screen_font_scale_default_comparison"]:
+                select_statement.append("NULL")
+            elif field.name == "fields_hash":
```

Comment:
Minor: you could add a small comment here, as mental note for future self, that in v2 the hash field was at a different level in the scheme.